kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -584,42 +600,13 @@ protected void handleTopicMetadataRequest(KafkaHeaderAndRequest metadataHar,
             // get all topics, filter by permissions.
             final Map<String, List<TopicName>> topicMap = new ConcurrentHashMap<>();
             getAllTopicsAsync().thenAccept((allTopicMap) -> {
-                if (allTopicMap.isEmpty()) {
-                    pulsarTopicsFuture.complete(allTopicMap);
-                    return;
-                }
-                AtomicInteger allTopicCount =
-                        new AtomicInteger(allTopicMap.values().stream().mapToInt(List::size).sum());
-                if (allTopicCount.get() == 0) {
-                    pulsarTopicsFuture.complete(topicMap);
-                    return;
-                }
-                final Runnable completeOne = () -> {
-                    if (allTopicCount.decrementAndGet() == 0) {
-                        pulsarTopicsFuture.complete(topicMap);
-                    }
-                };
-
-                allTopicMap.forEach((topic, list) -> {
-                   list.forEach((topicName ->
-                           authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.TOPIC, topicName.toString()))
-                           .whenComplete((authorized, ex) -> {
-                               if (ex != null || !authorized) {
-                                   allTopicMetadata.add(new TopicMetadata(
-                                           Errors.TOPIC_AUTHORIZATION_FAILED,
-                                           topic,
-                                           KopTopic.isInternalTopic(topicName.toString(), metadataNamespace),
-                                           Collections.emptyList()));
-                                   completeOne.run();
-                                   return;
-                               }
-                               topicMap.computeIfAbsent(
-                                       topic,
-                                       ignored -> Collections.synchronizedList(new ArrayList<>())
-                               ).add(topicName);
-                               completeOne.run();
-                           })));
-                });
+                allTopicMap.forEach((topic, list) ->
+                        list.forEach((topicName ->
+                                topicMap.computeIfAbsent(topic, ignored -> new ArrayList<>()).add(topicName))
+                        )
+                );
+                log.info("WK allTopicMap {}", allTopicMap);
```

Comment:
I think you used the Chinese initials of your name in the log, we should be able to delete this line of the log.